### ISSUE TITLE
Do you have the plan to opensource the training code?

### ISSUE THREAD

**Issue by zhangzhensong** [dwofk/fast-depth#3]
#2 
I am also interested whether will you release the training code?

**Comment by fangchangma** [comment]
We might release the training code in the future. For now, please refer to https://github.com/fangchangma/sparse-to-dense.pytorch for training.

**Comment by vivkr** [comment]
Hey @fangchangma and @dwofk,

Great job on the paper! I tried to understand how to train the network based on the information in the link above, but I didn't understand if I need to provide depth data to train the model. (Sorry if this is obvious, but I've looked through the code and was unable to figure this out.)
Thanks!

**Comment by dwofk** [comment]
Hi @vivkr,

If you are referring to providing sparse depth data to train the model, then no, you do not need to provide that. You can use the 'rgb' modality. The sparsifier and number of sparse depth samples become irrelevant.

Models in this project support 'RGB' input, not 'RGBd' input.

**Comment by HarryShen19950808** [comment]
Hi @dwofk and @fangchangma ,
I'm interested if I want to train my own dataset, is it possible to use the ground truth collected by ZED camera?
Thanks!

**Comment by HuynhLam** [comment]
Hi,
Can you specify some detail information for the training of the MobileNet-NNConv5 model?
Which loss function was used? 
How many epoch was run?
Thanks!

**Comment by dwofk** [comment]
Hi @HarryShen19950808 

Model definitions are provided [here](https://github.com/dwofk/fast-depth/blob/master/models.py). If you have your own dataset, you can try modifying the models as necessary and retraining for your particular dataset.

**Comment by dwofk** [comment]
Hi @HuynhLam 

Yes, the training for our models is based on the [sparse-to-dense](https://github.com/fangchangma/sparse-to-dense.pytorch) work.

* no, we use L1 loss (note that the cited sparse-to-dense work also uses L1 loss)
* yes, the data augmentation is the same
* yes, encoder weights are initialized using ImageNet-pretrained models
* yes, initial lr = 0.01
* yes, we train for 20 epochs

**Comment by rafiqul713** [comment]
Is it possible to train the model with KITTI odometry dataset (rgb and sparse gt depth)?
